src/backend/distributed/metadata/metadata_cache.c

```diff
@@ -307,73 +310,97 @@ bool
 IsCitusTableType(Oid relationId, CitusTableType tableType)
 {
 	CitusTableCacheEntry *tableEntry = LookupCitusTableCacheEntry(relationId);
+	if (tableEntry)
+	{
+		return IsCitusTableTypeCacheEntry(tableEntry, tableType);
+	}
 
-	/* we are not interested in postgres tables */
-	if (tableEntry == NULL)
+	if (CitusHasBeenLoaded())
 	{
+		/*
+		 * Even if Citus has been loaded, we couldn't find (or build) a cache
+		 * entry for relation. Then it's definitely a postgres table.
+		 */
 		return false;
 	}
-	return IsCitusTableTypeInternal(tableEntry, tableType);
+
+	/*
+	 * If we are issuing ALTER EXTENSION citus UPDATE, CitusHasBeenLoaded()
+	 * returns false and we cannot access/build cache entries. In that case,
+	 * we need to decide table type directly reading from pg_dist_partition
+	 * table.
+	 * Reading from pg_dist_partition wouldn't be as efficient as accessing
+	 * to cache, but this is the case only when upgrading/downgrading Citus
+	 * so we don't expect any performance critical code-paths to reach here.
+	 */
+	ereport(DEBUG4, (errmsg("will scan pg_dist_partition for relation with "
+							"OID=%u", relationId)));
+	return IsCitusTableTypeViaCatalog(relationId, tableType);
```

Comment:
Those two lines are not covered in this pr :/ using `IsCitusTableType` in https://github.com/citusdata/citus/pull/4174 will cover